### PR TITLE
add store iOS implement of image picker

### PR DESCRIPTION
**Pull request by pedia**
Since issue(flutter/flutter#10495) is applied, I try a little step of it.

This only add lib/image_picker.dart and implement of iOS.

Any advise is appreciated.

### CODE REVIEW

**Comment by mit-mit** [comment]
@pedia looks like your dart code has some analysis issues flagged by the travis job:
```
  info - Prefer final for variable declaration if reference is not reassigned at lib/image_picker.dart:21:11 - (prefer_final_locals)
```

File: packages/image_picker/lib/image_picker.dart [hunk 1]
```diff
@@ -15,4 +15,10 @@ class ImagePicker {
     final String path = await _channel.invokeMethod('pickImage');
     return new File(path);
   }
+
+  // Return bool for store
+  static Future<bool> store(String path) async {
+     bool flag = await _channel.invokeMethod('store', path);
```
Comment (collinjackson) [inline packages/image_picker/lib/image_picker.dart]: you can just return the result of invokeMethod and not make it async

File: packages/image_picker/ios/Classes/ImagePickerPlugin.m [hunk 2]
```diff
@@ -136,4 +141,17 @@ - (UIImage *)normalizedImage:(UIImage *)image {
   return normalizedImage;
 }
 
+- (void)storeFile:(NSString *)filePath {
+  NSData *data = [[NSData alloc] initWithContentsOfFile:filePath];
+  UIImage *image = [UIImage imageWithData:data];
+  UIImageWriteToSavedPhotosAlbum(image, nil, @selector(storeComplete), nil);
```
Comment (collinjackson) [inline packages/image_picker/ios/Classes/ImagePickerPlugin.m]: I'm not sure how this is supposed to work... Do you mean to say `UIImageWriteToSavedPhotosAlbum(image, self, @selector(storeComplete:didFinishSavingWithError:contextInfo:), nil)` here?

File: packages/image_picker/ios/Classes/ImagePickerPlugin.m [hunk 3]
```diff
@@ -136,4 +141,17 @@ - (UIImage *)normalizedImage:(UIImage *)image {
...
+}
+
+- (void)storeComplete:(UIImage *)image
+    didFinishSavingWithError:(NSError *)error
+                 contextInfo:(void *)contextInfo {
+                   NSLog(@"storeComplete");
```
Comment (collinjackson) [inline packages/image_picker/ios/Classes/ImagePickerPlugin.m]: We shouldn't log here.

File: packages/image_picker/ios/Classes/ImagePickerPlugin.m [hunk 4]
```diff
@@ -136,4 +141,17 @@ - (UIImage *)normalizedImage:(UIImage *)image {
...
+                   _result([NSNumber numberWithBool:TRUE]);
```
Comment (collinjackson) [inline packages/image_picker/ios/Classes/ImagePickerPlugin.m]: We should call result with an error here if the error isn't nil.

File: packages/image_picker/ios/Classes/ImagePickerPlugin.m [hunk 5]
```diff
@@ -68,6 +68,11 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     [alert addAction:library];
     [alert addAction:cancel];
     [_viewController presentViewController:alert animated:YES completion:nil];
+  }
+  else if([@"store" isEqualToString:call.method]) {
+    _result = result;
```
Comment (collinjackson) [inline packages/image_picker/ios/Classes/ImagePickerPlugin.m]: I don't think we want to trample over `_result` here, we should be able to pass result to `storeFile:` as an additional argument and then to `UIImageWriteToSavedPhotosAlbum` as the `contextInfo` argument

File: packages/image_picker/lib/image_picker.dart [hunk 6]
```diff
@@ -15,4 +15,10 @@ class ImagePicker {
     final String path = await _channel.invokeMethod('pickImage');
     return new File(path);
   }
+
+  // Return bool for store
```
Comment (collinjackson) [inline packages/image_picker/lib/image_picker.dart]: Comment should be /// (dartdoc) and could use improvement. We should make a note if it only works on iOS, or ideally implement Android at the same time.

I think this should be `Future<Null>` and complete with an error if the store fails.

**Comment by mit-mit** [comment]
@pedia did you get a chance to look at the review feedback?

**Comment by mit-mit** [comment]
Going to close this; please open a new PR with the incorporated feedback.

**Pull request closed without merging**